index.js

```diff
@@ -199,40 +196,42 @@ function filesToStream(filesPromise, options) {
 }
 
 /**
- * @param {Array<{entity: BemEntityName, level: String, tech: String, path: String}>} introspection - unordered file-entities list
- * @param {String[]} levels - ordered levels' paths list
- * @param {Tenorok[]} decl - resolved and ordered declaration
+ * @param {Object} opts - Options for harvester
+ * @param {Array<{entity: BemEntityName, level: String, tech: String, path: String}>} opts.introspection - unordered file-entities list
+ * @param {String[]} opts.levels - ordered levels' paths list
+ * @param {Object<String, String[]>} [opts.techMap] - deps techs to file techs mapper
+ * @param {BemCell[]} opts.decl - resolved and ordered declaration
  * @returns {Array<{entity: BemEntityName, level: String, tech: String, path: String}>} - resulting ordered file-entities list
  */
-function harvest(introspection, levels, decl/*: Array<{entity, tech}>*/) {
-    const hash = fileEntity => `${fileEntity.entity.id}.${fileEntity.tech}`;
-    const declIndex = _buildIndex(decl, hash);
-
-    const entityInIndex = file => declIndex[hash(file)] !== undefined;
-    return introspection
-        .filter(entityInIndex)
-        .filter(file => levels.indexOf(file.level) !== -1)
-        .sort((f1, f2) => hash(f1) === hash(f2)
-            ? levels.indexOf(f1.level) - levels.indexOf(f2.level)
-            : declIndex[hash(f1)] - declIndex[hash(f2)]);
+function harvest(opts) {
+    const hash = cell => `${cell.entity.id}.${cell.tech}`;
+    const declIndex = _buildIndex(opts.decl, hash);
+
+    const fileTechToDep = Object.keys(opts.techMap || {}).reduce((res, depTech) =>
+        (opts.techMap[depTech].forEach(fileTech => (res[fileTech] = depTech)), res),
+        {});
+
+    return opts.introspection
+        .filter(file => {
+            return declIndex[hash(file)] !== undefined ||
+                fileTechToDep[file.tech] && declIndex[hash(Object.assign({}, file, {tech: fileTechToDep[file.tech]}))] !== undefined;
+        })
+        .filter(file => opts.levels.indexOf(file.level) !== -1)
+        .sort((f1, f2) => f1.entity.id === f2.entity.id && (fileTechToDep[f1.tech] === fileTechToDep[f2.tech])
+            ? (opts.levels.indexOf(f1.level) - opts.levels.indexOf(f2.level))
+                || (opts.techMap[fileTechToDep[f1.tech]].indexOf(f1.tech) - opts.techMap[fileTechToDep[f1.tech]].indexOf(f2.tech))
+            : declIndex[f1.entity.id] - declIndex[f2.entity.id]);
 }
 
 /**
- * @param {Array<{entity: Tenorok, tech: String}>} list - List of tenoroks
+ * @param {BemCell[]} list - List of tenoroks
  * @param {Function} hash - Hashing function
  * @returns {Object<String, Number>} - Entity id to sort order
  */
 function _buildIndex(list, hash) {
-    return list.reduce((res, fileEntity, idx) => {
-        res[hash(fileEntity)] = idx;
+    return list.reduce((res, cell, idx) => {
+        res[cell.entity.id] = idx;
+        res[hash(cell)] = idx;
```

Comment:
why you need to provide `hash` function
it could be different?
could we move `hash` out of `harvest` as helper?